backend/deepchecks_monitoring/logic/data_ingestion.py

```diff
@@ -359,20 +359,29 @@ async def _send_with_retry(
         retry_count = 0
         while messages and retry_count < max_retries:
             send_futures = [
-                producer.send(topic_name, value=message, key=key)
+                await producer.send(topic_name, value=message, key=key)
                 for key, message in messages
             ]
 
             results = await asyncio.gather(*send_futures, return_exceptions=True)
 
             failed_messages = [
-                (key, message) for ((key, message), result) in zip(messages, results)
+                ((key, message), result) for ((key, message), result) in zip(messages, results)
                 if isinstance(result, Exception)
             ]
 
+            exception_types = {type(result).__name__ for _, result in failed_messages}
+            if len(exception_types) == 1:
+                self.logger.warning(f"All messages failed with the same error: {list(exception_types)[0]}")
```

Comment:
why not error?

Reply:
there's a retry and an error only if failed after retries